Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -4,20 +4,38 @@ class BTAnalyticsService: Equatable {
 
     // MARK: - Internal Properties
 
-    /// The HTTP client for communication with the analytics service endpoint. Exposed for testing.
-    var http: BTHTTP?
-    
     /// The FPTI URL to post all analytic events.
     static let url = URL(string: "https://api-m.paypal.com")!
 
+    /// The HTTP client for communication with the analytics service endpoint. Exposed for testing.
+    var http: BTHTTP?
+
+    /// Exposed for testing only
+    var shouldBypassTimerQueue = false
+
+    // MARK: - Private Properties
+
+    private static let events = BTAnalyticsEventsStorage()
+
     private let apiClient: BTAPIClient
+    /// Amount of time, in seconds, between batch API requests sent to FPTI
+    private let timerInterval = 20
```

Comment:
Take it or leave it:

I wouldn't mind going further down - maybe to 15 sec. It is still a large win for analytics. But, I have a fear of losing some metrics since we aren't doing any local persistence to encounter for backgrounding, app kill etc. _Also_ this timer gets set only after the first analytic event is triggered, _not_ when the SDK is initialized

Reply:
> Also this timer gets set only after the first analytic event is triggered, not when the SDK is initialized

With us now sending config events it does get triggered fairly close to initialization. The first event is now the `v1/config` call. That said, down to bump it down to 15s!

Reply:
Done: https://github.com/braintree/braintree_ios/pull/1295/commits/61aa5f16f8c8983f47b02cde6201309f01b330fd